src/Transactions/AccountTransactionPayload.cs

```diff
@@ -12,39 +14,107 @@ namespace Concordium.Sdk.Transactions;
+            AccountAddress.From(payload.Transfer.Receiver)
+        ),
+        PayloadCase.RegisterData => new RegisterData(
+            OnChainData.From(payload.RegisterData)
+        ),
+        PayloadCase.DeployModule => new DeployModule(
+            VersionedModuleSourceFactory.From(payload.DeployModule)
+        ),
+        PayloadCase.RawPayload => ParseRawPayload(payload.RawPayload),
+        PayloadCase.InitContract => throw new NotImplementedException(),
+        PayloadCase.UpdateContract => throw new NotImplementedException(),
+        PayloadCase.None => throw new NotImplementedException(),
+        _ => throw new MissingEnumException<PayloadCase>(payload.PayloadCase),
+    };
+
+    private static AccountTransactionPayload ParseRawPayload(Google.Protobuf.ByteString payload)
+    {
+        (AccountTransactionPayload?, string?) parsedPayload = (null, null);
+
+        switch ((TransactionType)payload.First())
+        {
+            case TransactionType.Transfer:
+            {
+                Transfer.TryDeserial(payload.ToArray(), out var output);
+                parsedPayload = output;
+                break;
+            }
+            case TransactionType.TransferWithMemo:
+            {
+                TransferWithMemo.TryDeserial(payload.ToArray(), out var output);
+                parsedPayload = output;
+                break;
+            }
+            case TransactionType.RegisterData:
+            {
+                RegisterData.TryDeserial(payload.ToArray(), out var output);
+                parsedPayload = output;
+                break;
+            }
+            case TransactionType.DeployModule:
+            {
+                DeployModule.TryDeserial(payload.ToArray(), out var output);
+                parsedPayload = output;
+                break;
+            }
+            case TransactionType.InitContract:
+            case TransactionType.Update:
+            case TransactionType.AddBaker:
+            case TransactionType.RemoveBaker:
+            case TransactionType.UpdateBakerStake:
+            case TransactionType.UpdateBakerRestakeEarnings:
+            case TransactionType.UpdateBakerKeys:
+            case TransactionType.UpdateCredentialKeys:
+            case TransactionType.EncryptedAmountTransfer:
+            case TransactionType.TransferToEncrypted:
+            case TransactionType.TransferToPublic:
+            case TransactionType.TransferWithSchedule:
+            case TransactionType.UpdateCredentials:
+            case TransactionType.EncryptedAmountTransferWithMemo:
+            case TransactionType.TransferWithScheduleAndMemo:
+            case TransactionType.ConfigureBaker:
+            case TransactionType.ConfigureDelegation:
+                parsedPayload = (new RawPayload(payload.ToArray()), null);
+                break;
+            default:
+                throw new MissingEnumException<TransactionType>((TransactionType)payload.First());
```

Comment:
I talked to Emil Holm, about this and the conclusion was no. I'll poke at it just to confirm though.

Reply:
I can confirm, `(TransactionType)100` does not throw an exception, and will also be caught by the `MissingEnumException`.

Reply:
Yes, casting an integer to an enum cannot cause an exception, but the value will instead just keep the integer representation https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum#conversions.

However, `payload.First()` will throw an exception if the payload is empty, but this would mean something went wrong in the node.